flaml/automl.py

```diff
@@ -456,30 +451,29 @@ def _validate_data(self, X_train_all, y_train_all, dataframe, label,
             label = 'y'
 
         if X_train_all is not None and y_train_all is not None:
-            if not (isinstance(X_train_all, np.ndarray) or issparse(X_train_all)
-                    or isinstance(X_train_all, pd.DataFrame)):
-                raise ValueError(
-                    "X_train_all must be a numpy array, a pandas dataframe, "
-                    "or Scipy sparse matrix.")
-            if not (isinstance(y_train_all, np.ndarray)
-                    or isinstance(y_train_all, pd.Series)):
-                raise ValueError(
-                    "y_train_all must be a numpy array or a pandas series.")
-            if X_train_all.size == 0 or y_train_all.size == 0:
-                raise ValueError("Input data must not be empty.")
+            assert (
```

Comment:
Generally `assert` is reserved for debugging or panic that must shutdown everything (see this [post](https://stackoverflow.com/questions/944592/best-practice-for-using-assert)). I think the old style is better -- it allows the user's program to catch and fix the error.

Reply:
An easy way to get around this is to replace `assert` with a custom function `check_value`, which raises a ValueError if the condition is false.

Reply:
Agree to have a better approach in future. For the record, using assert is not totally unreasonable: https://code-examples.net/en/q/e69d0.